string time
string c_mode
int64 num_p
p_info[] position
